### PR TITLE
Implement pow and refactor BinOp

### PR DESCRIPTION
**Pull request by Smit-create**
Reference code:
```py
def check():
    a: i32
    a = pow(2, 2)
```

ASR:

```bash
(TranslationUnit 
   (SymbolTable 
      1 
      {
         check: 
            (Subroutine 
               (SymbolTable 
                  2 
                  {
                     a: 
                        (Variable 
                           2 
                           a 
                           Local () () 
                           Default 
                           (Integer 4 []) 
                           Source 
                           Public 
                           Required .false.)
                  }) 
               check [] [
               (= 
                  (Var 2 a) 
                  (BinOp (ConstantInteger 2 
                     (Integer 4 [])) 
                     Pow (ConstantInteger 2 
                     (Integer 4 [])) 
                     (Integer 4 []) () ()) ())] 
               Source 
               Public 
               Implementation () .false. .false.)
      }) 
   [])
```

### CODE REVIEW

**Review by certik — APPROVED** [review]
I think this looks good. @namannimmo10, can you please also review this one?

File: src/lfortran/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -721,10 +705,32 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
...
-        tmp = ASR::make_BinOp_t(al, x.base.base.loc, left, op, right, dest_type,
+        tmp = ASR::make_BinOp_t(al, loc, left, op, right, dest_type,
                                 value, overloaded);
     }
 
+    void visit_BinOp(const AST::BinOp_t &x) {
+        this->visit_expr(*x.m_left);
+        ASR::expr_t *left = LFortran::ASRUtils::EXPR(tmp);
+        this->visit_expr(*x.m_right);
+        ASR::expr_t *right = LFortran::ASRUtils::EXPR(tmp);
```
Comment (namannimmo10) [inline src/lfortran/semantics/python_ast_to_asr.cpp:716]:
```suggestion
        ASR::expr_t *left = ASRUtils::EXPR(tmp);
        this->visit_expr(*x.m_right);
        ASR::expr_t *right = ASRUtils::EXPR(tmp);
```

**Comment by namannimmo10** [comment]
LGTM, too. 🚀 
We should resolve the conflicts and this can go in.

**Comment by Smit-create** [comment]
In it goes!

**Pull request merged**